### PR TITLE
feat:  use `setInitialState` for async initialization

### PR DESCRIPTION
**Pull request by amythos**
### usage
------
```javascript
module StateLenses = [%lenses
  type state = {
    description: string,
    title: string,
    acceptTerms: bool,
  }
];
module PostAddForm = ReFormNext.Make(StateLenses);

[@react.component]
let make = () => {
  let {state, submit, getFieldState, handleChange, setValues}: PostAddForm.api =
    PostAddForm.use(
      ~schema={
        PostAddForm.Validation.Schema([|
          Custom(
            Title,
            values =>
              Js.String.length(values.title) > 20
                ? Error("Keep it short!") : Valid,
          ),
          Custom(
            Description,
            values =>
              values.description == "" ? Error("Can't be empty") : Valid,
          ),
          Custom(
            AcceptTerms,
            values =>
              values.acceptTerms == false
                ? Error("You must accept all the terms") : Valid,
          ),
        |]);
      },
      ~onSubmit=
        ({state}) => {
          mutate(
            ~variables=
              PostAddMutationConfig.make(
                ~title=state.values.title,
                ~description=state.values.description,
                (),
              )##variables,
            (),
          )
          |> Js.Promise.then_(result =>
               setResult(_ => Some(result)) |> Js.Promise.resolve
             )
          |> ignore;

          None;
        },
      ~initialState={title: "", description: "", acceptTerms: false},
      (),
    );
  
  React.useEffect0(() => {
      fetchDefaultState(fetchUrl, state => setValues(state));
  });
};

```

### CODE REVIEW

**Comment by fakenickels** [comment]
Thanks, this makes a lot of sense!

File: packages/bs-reform/src/ReFormNext.re [hunk 1]
```diff
@@ -127,6 +147,7 @@ module Make = (Config: Config) => {
     arrayRemoveBy: 'a. (Config.field(array('a)), 'a => bool) => unit,
     submit: unit => unit,
     resetForm: unit => unit,
+    setInitialState: Config.state => unit,
```
Comment (fakenickels) [inline packages/bs-reform/src/ReFormNext.re]: I think we can actually create a helper called `setValues`

File: packages/bs-reform/src/ReFormNext.re [hunk 2]
```diff
@@ -144,6 +165,7 @@ module Make = (Config: Config) => {
     let (state, send) =
       ReactUpdate.useReducer(
         {
+          initValues: initialState,
```
Comment (fakenickels) [inline packages/bs-reform/src/ReFormNext.re]: With setValues I don't think this is necessary

Comment (fakenickels) [reply packages/bs-reform/src/ReFormNext.re]: hmm looking the ResetForm branch changes I got it now

File: packages/bs-reform/src/ReFormNext.re (hunk 1 above)
Comment (amythos) [reply packages/bs-reform/src/ReFormNext.re]: i think that setValues means set all values passed in, and setInitialState means won't set the value when its fieldState is not Pristine, maybe we can call it setinitialValues

Comment (fakenickels) [reply packages/bs-reform/src/ReFormNext.re]: that makes sense, I'll take a better look soon so we can get it merged

Comment (amythos) [reply packages/bs-reform/src/ReFormNext.re]: hello @fakenickels would you have time to merge this

Comment (fakenickels) [reply packages/bs-reform/src/ReFormNext.re]: Sorry about the delay, the past few days have we have been a bit busy. I'll gonna finish the review today and give you a better feedback soon

**Comment by fakenickels** [comment]
@amythos This looks good, but for the next releases we are going to try to make ReForm as similar as possible to Formik to ease adoption. So in Formik there is no setInitialValues, we just have a setValues. So before merging this PR we would like to have the handlers also named setValues and no `initValues` on the state, if the user needs to resetTheForm and apply the initialState the user will just call reset and call setValues as well.

File: packages/bs-reform/src/ReFormNext.re [hunk 3]
```diff
@@ -86,6 +86,24 @@ module Make = (Config: Config) => {
     ->Belt.Array.get(0);
   };
 
+  let setInitialValues =
```
Comment (fakenickels) [inline packages/bs-reform/src/ReFormNext.re]: let's call this setValues now

**Comment by amythos** [comment]
@fakenickels now we have setValues for set the whole state, and setFieldValue for the single field

**Comment by fakenickels** [comment]
Thank you for your time and effort @amythos , we'll be releasing this to NPM soon

**Comment by fakenickels** [comment]
@all-contributors please add @amythos for code

**Pull request merged**